scripts/test-check-packages.sh

```diff
@@ -62,8 +62,15 @@ for d in test/packages/${PACKAGE_TEST_TYPE:-other}/${PACKAGE_UNDER_TEST:-*}/; do
     cd $d
     elastic-package install -v
 
-    # defer-cleanup is set to a short period to verify that the option is available
-    elastic-package test -v --report-format xUnit --report-output file --defer-cleanup 1s --test-coverage
+    if [ "$(basename $d)" == "long_integers" ]; then
+      # Ensure that any change in unmarshaling behaviour is noticed; this will result in a dirty
+      # git state on exit if an inappropriate use of encoding/json.Unmarshal has been made.
+      elastic-package test -v -g --report-format xUnit --report-output file --defer-cleanup 1s --test-coverage
```

Comment:
Is the `-g` necessary here? Shouldn't it be caught due to differences compared to the `expected.json` file?

Reply:
+1, I think this if is not needed, `elastic-package test`  should fail if there are regressions in unmarshalling logic. If it doesn't fail, this is a different bug to be addressed in a different PR.

Reply:
No, it won't be caught. This was the reason for the additional code previously. The issue is that it is not possible for a machine to measure the accuracy of some thing that is more accurate than it, and this is what that would be asking it to do; if the the code is regressed, then it becomes a less accurate machine and so become inadequate to test that the test cases (which are more accurate) are satisfied. This is all theoretically sound, but can be practically confirmed by running the tests at 9b93a10 with this change
```diff
diff --git a/internal/testrunner/runners/pipeline/test_result.go b/internal/testrunner/runners/pipeline/test_result.go
index d67853c..38916f8 100644
--- a/internal/testrunner/runners/pipeline/test_result.go
+++ b/internal/testrunner/runners/pipeline/test_result.go
@@ -228,6 +228,7 @@ func unmarshalTestResult(body []byte) (*testResult, error) {
 // prevent low bit truncation of values greater than 1<<53.
 // See https://golang.org/cl/6202068 for details.
 func jsonUnmarshalUsingNumber(data []byte, v interface{}) error {
+       return json.Unmarshal(data, v)
        dec := json.NewDecoder(bytes.NewReader(data))
        dec.UseNumber()
        err := dec.Decode(v)
```
They will pass.

Reply:
Hmm.. to be honest, I'd accept that risk than introduce a sneaky workaround. It sounds to me like an edge case.

Reply:
Should I remove the tests? They essentially don't do anything without this.

Reply:
Let me challenge this. In this case, we're talking about the Go stdlib's nature, which unfortunately has this behavior. It means that we need to a test to detect a possible regression. I'm wondering if the unit test which calls `jsonUnmarshalUsingNumber`, then marshals back to `[]byte` using the right `json.Marshaller` can detect this issue. WDYT?

Reply:
The question that has to be asked is what does that test? Depending where it's put that could adequately test for all regressions, but as the code stands you will end up mocking essentially all of `(*runner).run` in order to achieve that.

There are many moving parts here, breaking some of them would be detected, but breaking others will not. The fact that breaking some of them will be is a good reason to keep the package here. I'm at the end of an extremely long day, so I don't have the cognitive capacity to sort out which are visible and which are not right now.

Reply:
Ok, I see, you make a good point with the comparison of floats.